src/BlockUniqidTransformer.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Class BlockUniqidTransformer.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use AMP_Block_Uniqid_Sanitizer;
+
+/**
+ * Transform uniqid-based IDs into cacheable IDs based on wp_unique_id.
+ *
+ * Random strings based on `uniqid()` should not be used to generate CSS class
+ * names and used in inline styles so that parsed CSS transient caching is not
+ * automatically disabled.
+ *
+ * Instead, `wp_unique_id()` should be used so that the IDs in the class names
+ * are more predictable and the CSS transient caching works as expected.
+ *
+ * @link https://github.com/ampproject/amp-wp/pull/6925
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.2.2
+ * @internal
+ */
+final class BlockUniqidTransformer implements Conditional, Service, Registerable {
+
+	/**
+	 * Check whether the Gutenberg plugin is present.
+	 *
+	 * @return bool
+	 */
+	public static function has_gutenberg_plugin() {
+		return defined( 'GUTENBERG_VERSION' );
+	}
+
+	/**
+	 * Check whether the conditional object is currently needed.
+	 *
+	 * @return bool Whether the conditional object is needed.
+	 */
+	public static function is_needed() {
+		return (
+			(
+				self::has_gutenberg_plugin()
+				||
+				version_compare( get_bloginfo( 'version' ), '5.9', '>=' )
```

Comment:
Once backported to core this will need to be updated. I suppose we could update it now to just be `6.0`:

```suggestion
				(
					version_compare( get_bloginfo( 'version' ), '5.9', '>=' )
					&&
					version_compare( get_bloginfo( 'version' ), '6.0', '<' )
				)
```

The `6.0` could be replaced with `5.9.x` when that lands.

Reply:
There's one more thing about WordPress versions. Duotone was [first released](https://make.wordpress.org/core/2021/07/03/wordpress-5-8-field-guide/) in `5.8` where it uses a slightly different selector pattern:

```html
<figure class="wp-duotone-filter-622b62d997e58 wp-block-image size-large">
	<img … />
</figure>
<style>
	.wp-duotone-filter-622b62d997e58 img {
		filter: url( #wp-duotone-filter-622b62d997e58 );
	}
</style>
<svg xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 0 0" width="0" height="0" focusable="false" role="none" style="visibility: hidden; position: absolute; left: -9999px; overflow: hidden;">
	<defs>
		<filter id="wp-duotone-filter-622b62d997e58">
			…
		</filter>
	</defs>
</svg>
```

It seems that we should sanitize this, too (the sanitized needs to also cover not only `wp-duotone-` but also `wp-duotone-filter-`).

Reply:
This has been addressed in 703e853 (WP versions check) and f68c8e2 (support legacy duotone prefix, i.e. `wp-duotone-filter-`).